File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include<iostream>
#include<string>

using namespace std;

int main() {

	string str;
	int sol = 0;
	int fe = 0;

	cin >> str;

	for (int i = 0; i < str.size(); i++) {
		if (str[i] == '(') {
			fe += 1;
		}
		else if (str[i] == ')') {
			if (str[i - 1] == '(') {
				fe -=1;
				sol += fe;
			}
			else {
				sol += 1;
				fe -= 1;
			}
		}
	}
	cout << sol;


	return 0;
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.io.*;
import java.util.Stack;

class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        String str = br.readLine();
        Stack<Character> st = new Stack<>();
        int count = 0;
        int sol = 0;

        for (int i=0; i<str.length(); i++){
            if (str.charAt(i)=='('){
                if(str.charAt(i+1)==')'){
                    sol += st.size();
                    i += 1;
                }
                else {
                    st.push(str.charAt(i));
                }
            }
            else {
                sol += 1;
                st.pop();
            }
        }

        bw.write(sol+"");

        bw.flush();
        bw.close();
    }
}